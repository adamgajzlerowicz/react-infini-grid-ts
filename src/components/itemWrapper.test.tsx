import * as React from 'react';
import * as renderer from 'react-test-renderer';

import ItemWrapper from './itemWrapper';

describe('ItemWrapper', () => {

  const Child = () => {
    return <div>blah</div>;
  };

  const component = renderer.create(<ItemWrapper height={250} itemsInRow={4} child={<Child />}/>);
  const tree = component.toJSON();
  
  it('renders', () => {
    expect(tree).toBeTruthy();
  });
  
  it('renders child', () => {
    expect(tree.children[0].children[0]).toEqual('blah');
  });

  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  it('sets correct styles from props', () => {
    expect(tree.props.style.flexBasis).toEqual(100 / 4 + '%');
    expect(tree.props.style.height).toEqual(250);
  });
  
});
