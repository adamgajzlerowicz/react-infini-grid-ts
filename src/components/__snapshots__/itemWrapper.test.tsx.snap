// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemWrapper matches the snapshot 1`] = `
<div
  className="item-wrapper"
  style={
    Object {
      "flexBasis": "25%",
      "flexGrow": 1,
      "height": 250,
    }
  }
>
  <div>
    blah
  </div>
</div>
`;
